conda/shell/etc/profile.d/conda.csh

```diff
@@ -27,16 +27,19 @@ if (`alias conda` == "") then
 else
     switch ( $1 )
         case "activate":
-            eval `$_CONDA_EXE shell.csh activate "$2" $argv[3-]`
+            set ask_conda="`(setenv prompt '${prompt}' ; '${_CONDA_EXE}' shell.csh activate '${2}' ${argv[3-]})`" || exit ${status}
```

Comment:
Just want to clarify... Is `exit` correct here?  In bash, it's be `return`, because we're running in the current process.  If we used `exit`, we'd be exiting out of the current shell, which isn't what we want.

Reply:
I do not have any one reference to prove this, rather it's easier to show you. Yay for spotty (T)CSH documentation...

I have included two scripts, one for (T)CSH and one for Bash. The Bash version is just a convenience that mirrors the (T)CSH script. Both scripts accepts one argument, the argument should be `s` (for source) or something else (for execute).

When a (T)CSH script is sourced the correct way to "bubble" out of the script is via `exit`, this will not kill your actual (T)CSH shell. If however the script is being executed, then `exec /bin/true` or `exec /bin/false` should be used (depending on the error code we want). I am not sure in what universe this made sense...

### Untitled.csh ((T)CSH Script)
```csh
./untitled.csh s  # an ok exit for non-nested scripts
./untitled.csh e  # a proper "bubble" out exit
source ./untitled.csh s  # a proper source exit
source ./untitled.csh e  # will kill top level shell
```

#### The Script
```csh
#!/usr/bin/env csh -f

# exit according to user inputs
if ( "${1}" == "s" ) then
    echo '(T)CSH Sourced - running `exit 1`'
    exit 1
else
    # get True value
    if ( -x "/usr/bin/false" ) then
        set FALSE=/usr/bin/false
    else
        set FALSE=/bin/false
    endif

    echo '(T)CSH Executed - running `exec '${FALSE}'`'
    exec ${FALSE}
endif
```

### Untitled.sh (Bash Script)
```bash
./untitled.sh s  # "can only `return' from a function or sourced script" error
./untitled.sh e  # a proper exit
. ./untitled.sh s  # a proper source exit
. ./untitled.sh e  # will kill top level shell
```

#### The Script
```bash
#!/usr/bin/env bash -f

# exit according to user inputs
if [ "${1}" == "s" ]; then
    echo 'BASH Sourced - running `return 1`'
    return 1
else
    echo 'BASH Executed - running `exit 1`'
    exit 1
fi
```

Reply:
lol cool.  That's way more than I needed.  I just wanted to double check and just be sure you really meant `exit`. I trust that you know (t)csh *way* better than I do.